packages/grid/_modules_/grid/models/colDef/gridColDef.ts

```diff
@@ -74,7 +74,7 @@ export interface GridColDef {
   /**
    * To be used in combination with `type: 'singleSelect'`. This is an array of the possible cell values and labels.
    */
-  valueOptions?: Array<any | { value: any; label: string }>;
+  valueOptions?: Array<string | number | { value: any; label: string }>;
```

Comment:
Can we use GridCellValue?

Reply:
I don't think we currently support `Date`

It will be considered as an object here https://github.com/mui-org/material-ui-x/pull/2112/files#diff-e03ef6ac8071c6dbcd6ceddc294a6ceb9692826571ae45332126f34b953d35efR8

Reply:
How would I do a single select for a list of country for example, with the flag and country name? 
I think the select options should consider renderCell, valueFormatter or getter...

Reply:
I'm not following you

Reply:
Well it's off topic: but I think this feature is a bit too basic at this stage and should consider renderCell, valueGetter, valueFormatter...

Reply:
This PR add the support of `number`, it's a step forward but I do agree that we still don't support everything we should

Reply:
This `singleSelect` is only a column type, like the others. The user can provide a custom cell or a different data structures.